src/middlewares/accessControl.js

```diff
@@ -0,0 +1,35 @@
+const { User } = require('../models/user');
+
+const isAdmin = async (req, res, next) => {
+  try {
+    const user = User.findById(req.user.id);
+    if (user.role === 'admin') {
+      next();
+    } else {
+      res.status(400).json({
+        status: 'Fail',
+        message: 'You are trying to access an admin route',
```

Comment:
For the messages, I think just showing a 404 error is enough without specifying what layer is in need, this is what I see generally on websites when I try to access unauthorized pages.

<img width="1183" alt="Screen Shot 2022-02-19 at 11 27 56 PM" src="https://user-images.githubusercontent.com/66186149/154822494-7d06d817-74e7-4ded-bc84-5b79f2a32732.png">

Reply:
Hey, @Kishimoto96. Could you clarify what _**without specifying what layer is in need**_ means? Also this 404 error is what you receive by trying to access a front-end page. It's not a request to API. What do you think about [403](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403)?

Reply:
Hi @ZuluNovember ,
I mean we don't have to let the user know which kind of authentication is needed to access any private page, and yeah just giving a 403 Forbidden error would be perfect, Thanks :)